### PR TITLE
Updating how SQL Server Identity Columns are treated

### PR DESCRIPTION
**Pull request by lajones**
@ErikEJ @natemcmaster 
Fix for #2465. There was a problem with how identity columns were being treated. Updated to produce ValueGeneratedNever() and ValueGeneratedOnAdd() correctly and updated the tests to test this.

### CODE REVIEW

**Comment by lajones** [comment]
I made E2E.sql a utf-8 file. That doesn't help the diff this time, but hopefully next time git will be able to successfully compare them.

File: test/EntityFramework.SqlServer.Design.FunctionalTests/ReverseEngineering/ExpectedResults/E2E/SqlServerReverseEngineerTestE2EContext.expected [hunk 1]
```diff
@@ -14,8 +14,6 @@ namespace E2ETest.Namespace
         {
             modelBuilder.Entity<AllDataTypes>(entity =>
             {
-                entity.Property(e => e.AllDataTypesID).ValueGeneratedNever();
-
```
Comment (lajones) [inline test/EntityFramework.SqlServer.Design.FunctionalTests/ReverseEngineering/ExpectedResults/E2E/SqlServerReverseEngineerTestE2EContext.expected]: I made this column be an IDENTITY column - so as to test the plain IDENTITY, integer column. That means that this property now is the single, integer, identity primary key mentioned in the comments and hence needs no fluent API.

File: test/EntityFramework.SqlServer.Design.FunctionalTests/ReverseEngineering/ExpectedResults/E2E/SqlServerReverseEngineerTestE2EContext.expected [hunk 2]
```diff
@@ -132,6 +130,10 @@ namespace E2ETest.Namespace
                 entity.Property(e => e.ReferredToByTableWithUnmappablePrimaryKeyColumnID).ValueGeneratedNever();
 
                 entity.Property(e => e.AColumn).Required();
+
+                entity.Property(e => e.ValueGeneratedOnAddColumn)
+                    .Required()
+                    .ValueGeneratedOnAdd();
             });
```
Comment (lajones) [inline test/EntityFramework.SqlServer.Design.FunctionalTests/ReverseEngineering/ExpectedResults/E2E/SqlServerReverseEngineerTestE2EContext.expected]: I added this as a test of a column which is an integer and is identity but is not the primary key.

**Comment by ErikEJ** [comment]
LGTM - Should this not be a PR to Dev?

**Comment by bricelam** [comment]
:shipit:

**Comment by lajones** [comment]
Checked in with commit 566da03

**Pull request closed without merging**